ap/production/features.py

```diff
@@ -0,0 +1,47 @@
+# coding: utf-8
+
+"""
+Column production methods related to higher-level features.
+"""
+
+from ap.production import producer
+from ap.util import maybe_import
+from ap.columnar_util import EMPTY, set_ak_column
+
+ak = maybe_import("awkward")
+
+
+def extract(ak_array: ak.Array, idx: int) -> ak.Array:
+    """
+    Takes an jagged *ak_array* and an index and returns padded array from given index.
+    """
+    ak_array = ak.pad_none(ak_array, idx + 1)
+    ak_array = ak.fill_none(ak_array[:, idx], EMPTY)
+    return ak_array
+
+
+@producer(
+    uses={
+        "Electron.pt", "Electron.eta", "Muon.pt", "Muon.eta", "Jet.pt", "Jet.eta",
+        "Jet.btagDeepFlavB",
+    },
+    produces={"HT", "nJet", "nElectron", "nMuon", "nDeepjet", "Electron1_pt", "Muon1_pt"} | {
+        f"Jet{i}_{attr}"
+        for i in range(1, 4 + 1)
+        for attr in ["pt", "eta"]
+    },
+)
+def variables(events: ak.Array, **kwargs) -> ak.Array:
```

Comment:
This is an optional comment: I realize that this producer will probably be something analysis-dependent and should therefore not be part of the analysis playground core, but it might be nice to add a doc string to producers that explains what columns the producer creates. I propose to already start this in this early stage as future reference for other people. 

Now that I think about it, it might be beneficial to have a producer/producers that compute common variables that can be used in many analyses, e.g. HT 🤔

Reply:
I think a set of default or exemplary producers/selectors/variables would be nice to 'quickstart' an analysis / get to know the framework for newcomers

Reply:
Yup, second it. At some point we're going to strip away the analysis-dependent implementations but in the meantime, it's nice to be able to plot sth that is not there from there start :)

In the long term, we should of course enforce doc strings as @pkausw suggests, especially for "core" stuff.